File: src/Undersoft.SharedServiceCenter.Web.Application.React/src/undersoft.ssc.web.application.client/src/account/identity/accountIdentityCredentials.ts
import { Model } from "@/models/model";

export interface AccountIdentityCredentials extends Model {
    UserName?: string;
    Email?: string;
    Password?: string;
    PhoneNumber?: string;
    EmailConfirmed?: boolean;
    PhoneNumberConfirmed?: boolean;
    SessionToken?: string;
    ResetPasswordToken?: string;
    EmailConfirmationToken?: string;
    PhoneNumberConfirmationToken?: string;
}

File: src/Undersoft.SoftwareDevelopmentKit/src/Undersoft.SDK.Service.Application/Account/Identity/AccountIdentityCredentials.cs
﻿using Microsoft.AspNetCore.Identity;
using Undersoft.SDK.Security.Identity;

namespace Undersoft.SDK.Service.Application.Account.Identity;

public class AccountIdentityCredentials : Credentials
{
    public AccountIdentityCredentials() { }

    public void MapUser(IdentityUser<long> account)
    {
        UserName = account.UserName;
        Email = account.Email;
        PhoneNumber = account.PhoneNumber;
        EmailConfirmed = account.EmailConfirmed;
        PhoneNumberConfirmed = account.PhoneNumberConfirmed;
        AccessFailedCount = account.AccessFailedCount;
        NormalizedUserName = account.NormalizedUserName;
    }
}
